pkg/trace/agent/run.go

```diff
@@ -168,3 +173,30 @@ func Run(ctx context.Context) {
 		f.Close()
 	}
 }
+
+// runningProfiling runs the profiler when it is enabled and reports whether
+// it did so.
+func runningProfiling(cfg *config.AgentConfig) bool {
```

Comment:
Nit:
I find the naming a little confusing, like it's a *check* rather than something that's going to start the profiler.
Maybe `runProfiling` or `startProfiling` or such.

Reply:
Cool. Check latest commit.